plugins/inputs/netdata/netdata.go

```diff
@@ -0,0 +1,291 @@
+
+  ## Should measurement be prefixed by the netdata hostname ?
+  ## Ex : netdata.host1.apps.cpu (with hostname),  netdata.apps.cpu (without hostname)
+  ## default to true
+  prefixChartIDByHostname = true
+
+  ## Netdata servers
+  [[inputs.netdata.servers]]
+    url="http://netdata-host:19999"
+
+  [[inputs.netdata.servers]]
+    url="https://netdata-host2"
+    basicAuthUsername="user"
+    basicAuthPassword="pass"
+`
+
+type Netdata struct {
+	Servers                 []Server
+	Points                  int
+	Group                   string
+	PrefixChartIDByHostname bool
+
+	client *http.Client
+}
+
+type Server struct {
+	Url               string
+	BasicAuthUsername string
+	BasicAuthPassword string
+	lastGatherTime    int64
+	currGatherTime    int64
+}
+
+type NestedError struct {
+	Err       error
+	NestedErr error
+}
+
+func (ne NestedError) Error() string {
+	return ne.Err.Error() + ": " + ne.NestedErr.Error()
+}
+
+func Errorf(err error, msg string, format ...interface{}) error {
+	return NestedError{
+		NestedErr: err,
+		Err:       fmt.Errorf(msg, format...),
+	}
+}
+
+type ResponseCharts struct {
+	Hostname string           `json:"hostname"`
+	Charts   map[string]Chart `json:"charts"`
+}
+
+type Chart struct {
+	ID         string                    `json:"id"`
+	Type       string                    `json:"type"`
+	Family     string                    `json:"family"`
+	DataURL    string                    `json:"data_url"`
+	Dimensions map[string]ChartDimension `json:"dimensions"`
+}
+
+type ChartDimension struct {
+	Name string `json:"name"`
+}
+
+type ResponseData struct {
+	DimensionNames []string   `json:"dimension_names"`
+	Result         DataResult `json:"result"`
+}
+
+type DataResult struct {
+	Data [][]interface{} `json:"data"`
+}
+
+func init() {
+	inputs.Add("netdata", func() telegraf.Input {
+		return &Netdata{
+			Points: 10,
+			Group:  "average",
+			PrefixChartIDByHostname: true,
+		}
+	})
+}
+
+func (n *Netdata) Description() string {
+	return description
+}
+
+func (n *Netdata) SampleConfig() string {
+	return sampleConfig
+}
+
+func (n *Netdata) Gather(acc telegraf.Accumulator) error {
+	var wg sync.WaitGroup
+
+	if n.client == nil {
+		tr := &http.Transport{
+			ResponseHeaderTimeout: time.Duration(3 * time.Second),
+		}
+		n.client = &http.Client{
+			Transport: tr,
+			Timeout:   time.Duration(4 * time.Second),
+		}
+	}
+
+	for idx := range n.Servers {
+		wg.Add(1)
+		go func(server *Server) {
+			defer wg.Done()
+			n.gatherServer(acc, server)
+		}(&n.Servers[idx])
+	}
+
+	wg.Wait()
+	return nil
+}
+
+func (n *Netdata) gatherServer(acc telegraf.Accumulator, server *Server) {
+	server.currGatherTime = time.Now().Unix()
+
+	// first time gather data from the past 10s
+	if server.lastGatherTime == 0 {
+		server.lastGatherTime = time.Now().Unix() - 10
+	}
+
+	chartRequestURL, err := url.Parse(server.Url + chartsURL)
+	if err != nil {
+		acc.AddError(Errorf(err, "error parsing charts url : %s", server.Url+chartsURL))
```

Comment:
How about
```suggestion
		acc.AddError(fmt.Errorf("error parsing charts url '%s': %s", server.Url+chartsURL, err))
```
Same for the instances below. This makes the whole _NestedError_ part obsolete I guess.

Reply:
Nice to have.